oshi-core/src/main/java/oshi/util/platform/windows/RegistryUtil.java

```diff
@@ -0,0 +1,181 @@
+        if (val instanceof Integer) {
+            int value = (Integer) val;
+            if (value > minSaneTimestamp && value < currentTimeSecs) {
+                return value * 1000L;
+            }
+            return value;
+        } else if (val instanceof String) {
+            String dateStr = ((String) val).trim();
+            // Try yyyyMMdd first
+            long epoch = ParseUtil.parseDateToEpoch(dateStr, "yyyyMMdd");
+            if (epoch == 0) {
+                // If that fails, try MM/dd/yyyy
+                epoch = ParseUtil.parseDateToEpoch(dateStr, "MM/dd/yyyy");
+            }
+            return epoch;
+        }
+        return 0L;
+    }
+
+    /**
+     * Returns a registry value as a String. (without access flag) Currently supports String and Binary
+     */
+    public static String getStringValue(WinReg.HKEY root, String path, String key) {
+        try {
+            Object val = Advapi32Util.registryGetValue(root, path, key);
+            return registryValueToString(val);
+        } catch (Win32Exception e) {
+            LOG.trace("Unable to access " + path + ": " + e.getMessage());
+            return null;
+        }
+    }
+
+    /**
+     * Returns a registry value as a String. (with access flag) Currently supports String and Binary
+     */
+    public static String getStringValue(WinReg.HKEY root, String path, String key, int accessFlag) {
+        Object val = getRegistryValueOrNull(root, path, key, accessFlag);
+        return registryValueToString(val);
+    }
+
+    /**
+     * Decodes registry value to String using multiple fallback encodings.
+     */
+    private static String registryValueToString(Object val) {
+        if (val == null)
+            return null;
+
+        // Already a string (REG_SZ or REG_EXPAND_SZ)
+        if (val instanceof String)
+            return ((String) val).trim();
+
+        // handle binary (REG_BINARY)
+        if (val instanceof byte[]) {
+            return decodeBinaryToString((byte[]) val);
+        }
+
+        return null;
+    }
+
+    /**
+     * Attempts to decode REG_BINARY into a string, trying UTF-16LE → UTF-8 → Windows-1252 → Hex fallback.
+     */
+    private static String decodeBinaryToString(byte[] bytes) {
+        if (bytes == null || bytes.length == 0)
+            return null;
+
+        String[] encodings = { "UTF-16LE", "UTF-8", "Windows-1252" };
+
+        for (String enc : encodings) {
+            try {
+                String decoded = new String(bytes, enc).trim();
+                if (!decoded.isEmpty()) {
+                    return decoded;
+                }
+            } catch (UnsupportedEncodingException e) {
+            }
+        }
+
+        // fall back to Hex
+        StringBuilder sb = new StringBuilder();
+        for (byte b : bytes) {
+            sb.append(String.format(Locale.ROOT, "%02X", b));
+        }
+
+        return sb.length() == 0 ? null : sb.toString();
```

Comment:
```suggestion
        return Arrays.stream(bytes)
            .mapToObj(b -> String.format(Locale.ROOT, "%02X", b))
            .collect(Collectors.joining(" "));
```
More readable with a space.